diff --git a/escapegame/management/commands/populate-database.py b/escapegame/management/commands/populate-database.py
index 9e2cd19..1a489fb 100644
--- a/escapegame/management/commands/populate-database.py
+++ b/escapegame/management/commands/populate-database.py
@@ -188,6 +188,11 @@ class Command(BaseCommand):
 		raspi_son_stranger_things = RaspberryPi(name='Sons Stranger Things', hostname='sons-stranger-things.local', media_type= "audio")
 		raspi_son_stranger_things.save()
 
+		# Just for debug so that everything points to master
+		raspi_1001_nuits = raspi_master
+		raspi_sons_1001_nuits = raspi_master
+		raspi_stranger_things = raspi_master
+		raspi_son_stranger_things = raspi_master
 
 		self.stdout.write(self.style.SUCCESS(' OK'))
 
diff --git a/scripts/init-database-mysql.sh b/scripts/init-database-mysql.sh
index fc582ce..53ae431 100755
--- a/scripts/init-database-mysql.sh
+++ b/scripts/init-database-mysql.sh
@@ -139,8 +139,8 @@ MASTER(){
 }
 
 
-if [ $HOSTNAME == $MASTER_HOSTNAME ]; then
+#if [ $HOSTNAME == $MASTER_HOSTNAME ]; then
 	MASTER
-fi
+#fi
 
 CREATE_MYSQL_CLIENT_CONFIG "${MASTER_HOSTNAME}"
diff --git a/siteconfig/settings.py b/siteconfig/settings.py
index f03f00c..d775832 100644
--- a/siteconfig/settings.py
+++ b/siteconfig/settings.py
@@ -149,7 +149,7 @@ DATABASE_MYSQL = {
 	'NAME': 'escapegame',
 	'USER': 'escapegame',
 	'PASSWORD': 'escapegame',
-	'HOST': MASTER_HOSTNAME,
+	'HOST': 'localhost',
 	'OPTIONS': {
 		'init_command': "SET sql_mode='STRICT_TRANS_TABLES'",
 	},
@@ -233,7 +233,8 @@ WEBSOCKET_URL = '/ws/'
 # Use Redis as the session engine
 # https://django-websocket-redis.readthedocs.io/en/latest/installation.html#replace-memcached-with-redis
 
-REDIS_HOST = MASTER_HOSTNAME
+#REDIS_HOST = MASTER_HOSTNAME
+REDIS_HOST = 'localhost'
 REDIS_PORT = 6379
 
 SESSION_REDIS_PREFIX = 'session'
